### DISCUSSION TITLE
How to get y position on tx ultimate touch?

### DISCUSSION THREAD

**Discussion by waynewcwu** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#61] [Q&A]
Hi,

I konw we can get x position at byte[5] or byte[6], as below code in Tx_Ultimate_Touch_.cpp .

May I ask how to **get y position ?** 

thanks.

```
int TxUltimateTouch::get_x_touch_position(int bytes[])
        {
            int state = bytes[4];
            switch (state)
            {
            case TOUCH_STATE_RELEASE:
                return bytes[5];
                break;

            case TOUCH_STATE_ALL_FIELDS:
                return bytes[5];
                break;

            case TOUCH_STATE_SWIPE_LEFT:
                return bytes[5];
                break;

            case TOUCH_STATE_SWIPE_RIGHT:
                return bytes[5];
                break;

            default:
                return bytes[6];
                break;
            }
        }
```

**Comment by barth11** [comment]
I don't think y axis is relevant because the touch stripes are one from top to bottom. It doesn't matter if you touch 2 cm from top or in the middle. It just activates the whole stripe. So x axis is enough to know where it was pressed.
Here snap from Daniel's video:
![{C315E18D-96C9-4EE3-8BB8-6D5347B5CF38}](https://github.com/user-attachments/assets/4f1781df-32b5-4322-b5a2-cff1cb39f878)
Timestamped link: https://youtu.be/naDLhX89enQ?si=C6tcnrCzP3WmecYG&t=311
